### PR TITLE
[DS-148] AutoComplete and some Input improvements

### PR DESCRIPTION
**Pull request by allyssonsantos**
This PR adds the new `<AutoComplete />` to web platform and do some improvements in `<Input />` component.

autocomplete

The `<Input />` improvements are:
- There's no more a `background-color: #fff;` at the input label, so it's possible to use it at any background.
- Change the `onClean` behavior, now it just return the cleaned value for the user handle it.

input

### CODE REVIEW

File: packages/yoga/src/Input/web/Field.jsx [hunk 1]
```diff
@@ -11,11 +11,8 @@ const labelTransition = css`
       },
     },
   }) => `
-    top: -1px;
-    left: ${input.padding.left - 2}px;
-
-    padding-right: ${input.label.padding.right}px;
-    padding-left: ${input.label.padding.left}px;
+    top: 0px;
```
Comment (ggdaltoso) [inline packages/yoga/src/Input/web/Field.jsx]:
```suggestion
    top: 0;
```

File: packages/yoga/src/Input/web/Legend.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import styled from 'styled-components';
+import { bool, node } from 'prop-types';
+
+const StyledLegend = styled.legend`
+  visibility: hidden;
```
Comment (ggdaltoso) [inline packages/yoga/src/Input/web/Legend.jsx]: CSSCOMB

File: packages/yoga/src/Input/web/Wrapper.jsx [hunk 3]
```diff
@@ -20,8 +21,32 @@ const Wrapper = styled.div`
   }) => `
     height: ${input.height}px;
 
-    border-radius: ${input.border.radius}px;
-    border: ${input.border.width}px solid ${input.border.color.default};
+    ${
+      !label
+        ? `
+        border-radius: ${input.border.radius}px;
+        border: 1px solid #d3d3e2;
```
Comment (ggdaltoso) [inline packages/yoga/src/Input/web/Wrapper.jsx]: Should be this border inside the theme?

File: packages/yoga/src/Input/web/Wrapper.jsx [hunk 4]
```diff
@@ -45,36 +70,52 @@ const Wrapper = styled.div`
       cursor: pointer;
     }
 
-    & {
-      width: ${full ? '100%' : `${input.width}px`};
+    &:focus-within {
+      legend {
+        max-width: 1000px;
+        font-weight: 700;
```
Comment (ggdaltoso) [inline packages/yoga/src/Input/web/Wrapper.jsx]: should be this prop inside the theme?

File: packages/yoga/src/Theme/themes/BaseTheme.js [hunk 5]
```diff
@@ -109,6 +110,15 @@ const BaseTheme = ({ primary, secondary, tertiary }) => {
       fontSizes,
       fontWeights,
     }),
+    autoComplete: autocomplete({
```
Comment (ggdaltoso) [inline packages/yoga/src/Theme/themes/BaseTheme.js]:
```suggestion
    autocomplete: autocomplete({
```

Comment (allyssonsantos) [reply packages/yoga/src/Theme/themes/BaseTheme.js]: done

File: packages/yoga/src/Input/web/Wrapper.jsx (hunk 4 above)
Comment (allyssonsantos) [reply packages/yoga/src/Input/web/Wrapper.jsx]: done

File: packages/yoga/src/Input/web/Wrapper.jsx (hunk 3 above)
Comment (allyssonsantos) [reply packages/yoga/src/Input/web/Wrapper.jsx]: done

File: packages/yoga/src/Input/web/Legend.jsx (hunk 2 above)
Comment (allyssonsantos) [reply packages/yoga/src/Input/web/Legend.jsx]: done

File: packages/yoga/src/Input/web/Field.jsx (hunk 1 above)
Comment (allyssonsantos) [reply packages/yoga/src/Input/web/Field.jsx]: done

File: packages/yoga/src/AutoComplete/web/AutoComplete.jsx [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+                      index: optionIndex,
+                      item: option,
+                      selected: highlightedIndex === optionIndex,
+                    })}
+                  >
+                    {option
+                      .split(reg)
+                      .map((part, index) =>
+                        part.match(reg) ? (
+                          <Match key={`${index}`}>{part}</Match>
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply packages/yoga/src/AutoComplete/web/AutoComplete.jsx]: done

File: packages/yoga/src/AutoComplete/web/AutoComplete.jsx [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+    },
+  }) => `
+    top: ${autoComplete.height}px;
+
+    width: ${full ? '100%' : `${autoComplete.width}px`};
+    max-height: ${autoComplete.height * 6}px;
+
+    background-color: ${autoComplete.field.backgroundColor};
+    border:
+      ${autoComplete.border.width}px solid ${autoComplete.border.color.typed};
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply packages/yoga/src/AutoComplete/web/AutoComplete.jsx]: done

**Pull request merged**